sql_runner/snowflake_target.go

```diff
@@ -122,9 +138,16 @@ func (sft SnowFlakeTarget) RunQuery(query ReadyQuery, dryRun bool, showQueryOutp
 		} else {
 			res, err := sft.Client.ExecContext(ctx, script)
 			if err != nil {
-				return QueryStatus{query, query.Path, int(affected), err}
+				switch err.Error() {
+				// If the error message is `-00001: `, the DB failed to return accurate status. Request the status and proceed accordingly.
+				case "-00001: ":
```

Comment:
Stirng matching against an error makes me a little uneasy, particularly one so generic. However, I do appreciate our options are limited here.
I guess this string comes from the driver library rather than Snowflake itself, so if it changes in the future it would only need to change here if we bump the snowflake driver? i.e. a change could only be caused by us upgrading the library?

Reply:
I feel the same. You are correct, it comes from the driver, which produces this error when the DB returns an empty response with no error. Stictly, it's possible for the DB to produce this if it sends back an empty error message with a status code of `-1`. However this is unlikely, and even in this scenario the desired behaviour would be to handle it as we handle it in this PR.

The other option was to skip the `sql` package and access the driver API directly when we're making calls to execute queries. That would allow us to switch on the status code itself. I experimented with that way, but it felt equally as jank if not more, and since I don't well understand the implications of skipping the `sql` packages handlers, I figured 'better the devil you know'. None of the documentation makes reference to directly accessing the driver like that.

Reply:
Is there a possibility that, we get this -1 code back from the driver (e.g. maybe a timeout or something), but the warehouse is still running the query and we then move on to the next step when the previous one hasn't finished?

Reply:
In that scenario, when we poll for the result, we'll get an error back that suggests the query is still running. We handle that on line 228 - we just loop until we get the actual result.